airbyte-integrations/connectors/destination-pubsub/src/test-integration/java/io/airbyte/integrations/destination/pubsub/PubsubDestinationAcceptanceTest.java

```diff
@@ -0,0 +1,250 @@
+import com.google.cloud.pubsub.v1.stub.GrpcSubscriberStub;
+import com.google.cloud.pubsub.v1.stub.SubscriberStub;
+import com.google.cloud.pubsub.v1.stub.SubscriberStubSettings;
+import com.google.common.collect.ImmutableMap;
+import com.google.pubsub.v1.AcknowledgeRequest;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PullRequest;
+import com.google.pubsub.v1.PullResponse;
+import com.google.pubsub.v1.ReceivedMessage;
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.string.Strings;
+import io.airbyte.integrations.base.AirbyteStreamNameNamespacePair;
+import io.airbyte.integrations.base.JavaBaseConstants;
+import io.airbyte.integrations.standardtest.destination.DestinationAcceptanceTest;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PubsubDestinationAcceptanceTest extends DestinationAcceptanceTest {
+
+  private static final Logger LOGGER = LoggerFactory
+      .getLogger(PubsubDestinationAcceptanceTest.class);
+  private static final Path CREDENTIALS_PATH = Path.of("secrets/credentials.json");
+  private TopicAdminClient topicAdminClient;
+  private SubscriptionAdminClient subscriptionAdminClient;
+  private TopicName topicName;
+  private ProjectSubscriptionName subscriptionName;
+  private Credentials credentials;
+  private JsonNode configJson;
+  // TODO: is this ok for a streaming destination test?
+  private List<JsonNode> records;
+
+  @Override
+  protected String getImageName() {
+    return "airbyte/destination-pubsub:dev";
+  }
+
+  @Override
+  protected JsonNode getConfig() {
+    return configJson;
+  }
+
+  @Override
+  protected JsonNode getFailCheckConfig() {
+    ((ObjectNode) configJson).put(CONFIG_PROJECT_ID, "fake");
+    return configJson;
+  }
+
+  @Override
+  protected boolean implementsNamespaces() {
+    return true;
+  }
+
+  private AirbyteStreamNameNamespacePair fromJsonNode(JsonNode j) {
+    var stream = j.get(STREAM).asText("");
+    var namespace = j.get(NAMESPACE).asText("");
+    return new AirbyteStreamNameNamespacePair(stream, namespace);
+  }
+
+  @Override
+  protected List<JsonNode> retrieveRecords(TestDestinationEnv testEnv,
+                                           String streamName,
+                                           String namespace,
+                                           JsonNode streamSchema)
+      throws IOException {
+    if (records.isEmpty()) {
+      // first time retrieving records, retrieve all and keep locally for easier
+      // verification
+      SubscriberStubSettings subscriberStubSettings =
+          SubscriberStubSettings.newBuilder()
```

Comment:
Tests were failing locally for me until I added this line
```suggestion
          SubscriberStubSettings.newBuilder()
          .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
```